packages/compile-solidity/compilerSupplier/loadingStrategies/Docker.js

```diff
@@ -29,20 +42,45 @@ class Docker extends LoadingStrategy {
       };
     } catch (error) {
       if (error.message === "No matching version found") {
-        throw this.errors("noVersion", versionString);
+        throw new NoVersionError(versionString);
       }
       throw new Error(error);
     }
   }
 
-  getDockerTags() {
-    return axios.get(this.config.dockerTagsUrl, { maxRedirects: 50 })
-      .then(response => response.data.results.map(item => item.name))
-      .catch(error => {
-        throw this.errors("noRequest", this.config.dockerTagsUrl, error);
-      });
+  async list() {
+    const allTags = this.streamAllDockerTags();
+
+    // split stream of all tags into separate releases and prereleases streams
+    const isRelease = name => !!semver.valid(name);
```

Comment:
I will as always suggest using `Boolean(x)` instead of `!!x`. :)

Reply:
Let's see what [my Twitter poll](https://twitter.com/gnidan/status/1425960578049265674) says

Reply:
:P